account_payment_term_extension/models/account_payment_term_line.py

```diff
@@ -0,0 +1,157 @@
+
+from odoo import _, api, exceptions, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import date_utils
+from odoo.tools.float_utils import float_round
+
+
+class AccountPaymentTermLine(models.Model):
+    _inherit = "account.payment.term.line"
+
+    amount_round = fields.Float(
+        string="Amount Rounding",
+        digits="Account",
+        help="Sets the amount so that it is a multiple of this value.",
+    )
+    delay_type = fields.Selection(
+        selection=lambda self: self._get_delay_type(),
+        ondelete={
+            "weeks_after": "set default",
+            "weeks_after_end_of_month": "set default",
+            "weeks_after_end_of_next_month": "set default",
+            "months_after": "set default",
+            "months_after_end_of_month": "set default",
+        },
+    )
+    value = fields.Selection(
+        selection_add=[
+            ("percent_amount_untaxed", "Percent (Untaxed amount)"),
+            ("fixed",),
+        ],
+        ondelete={"percent_amount_untaxed": lambda r: r.write({"value": "percent"})},
+    )
+
+    payment_days = fields.Char(
+        string="Payment day(s)",
+        help="Put here the day or days when the partner makes the payment. "
+        "Separate each possible payment day with dashes (-), commas (,) "
+        "or spaces ( ).",
+    )
+
+    def _get_delay_type(self):
+        delay_setting = self.env.company.payment_terms_delay_type
+        payment_terms_delay_type = delay_setting or False
+        date_type = [
+            ("days_after", "Days after invoice date"),
+            ("days_after_end_of_month", "Days after end of month"),
+            ("days_after_end_of_next_month", "Days after end of next month"),
+            ("days_end_of_month_on_the", "Days end of month on the"),
+        ]
+        if payment_terms_delay_type:
+            if payment_terms_delay_type == "months":
+                date_type += [
+                    ("months_after", "Months after invoice date"),
+                    ("months_after_end_of_month", "Months after end of month"),
+                ]
+            if payment_terms_delay_type == "weeks":
+                date_type += [
+                    ("weeks_after", "Weeks after invoice date"),
+                    ("weeks_after_end_of_month", "Weeks after end of month"),
+                    ("weeks_after_end_of_next_month", "Weeks after end of next month"),
+                ]
+            if payment_terms_delay_type == "weeks_and_months":
+                date_type += [
+                    ("weeks_after", "Weeks after invoice date"),
+                    ("weeks_after_end_of_month", "Weeks after end of month"),
+                    ("weeks_after_end_of_next_month", "Weeks after end of next month"),
+                    ("months_after", "Months after invoice date"),
+                    ("months_after_end_of_month", "Months after end of month"),
+                ]
+        return date_type
```

Comment:
```suggestion
        weeks_delay_type = [
            ("weeks_after", "Weeks after invoice date"),
            ("weeks_after_end_of_month", "Weeks after end of month"),
            ("weeks_after_end_of_next_month", "Weeks after end of next month"),
        ]
        months_delay_type = [
            ("months_after", "Months after invoice date"),
            ("months_after_end_of_month", "Months after end of month"),
        ]
        if payment_terms_delay_type:
            if payment_terms_delay_type == "months":
                date_type += months_delay_type
            elif payment_terms_delay_type == "weeks":
                date_type += weeks_delay_type
            elif payment_terms_delay_type == "weeks_and_months":
                date_type += weeks_delay_type + months_delay_type
        return date_type
```
I believe this helps improve the code readability for this section. WDYT?